aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -257,19 +281,35 @@ Tensor stft(const Tensor& self, const int64_t n_fft, const optional<int64_t> hop
   if (window_.defined()) {
     input = input.mul(window_);
   }
-  // rfft and transpose to get (batch x fft_size x num_frames)
-  auto out = input.rfft(1, normalized, onesided).transpose_(1, 2);
+
+  // FFT and transpose to get (batch x fft_size x num_frames)
+  const bool complex_fft = input.is_complex();
```

Comment:
does this also need to check if the window is complex? Right below this the torch check implies it does.

Reply:
`input` here is the windowed view of the `self` tensor. A few lines above this,
```cpp
if (window_.defined()) {
  input = input.mul(window_);
}
```

So `input` will be complex if either `self` or `window` is complex.